pkg/controller/flow/reconcile.go

```diff
@@ -342,3 +324,22 @@ func (r *reconciler) getDefaultClusterBusName() (string, error) {
 
 	return fallbackClusterBusName, nil // return the fallback value
 }
+
+func (r *reconciler) getFeedForFlow(flow *v1alpha1.Flow) (*feedsv1alpha1.Feed, error) {
+	feedList := &feedsv1alpha1.FeedList{}
+	err := r.client.List(
+		context.TODO(),
+		&client.ListOptions{Namespace: flow.Namespace, LabelSelector: labels.Everything()},
+		feedList)
+	if err != nil {
+		glog.Errorf("Unable to list feeds: %v", err)
+		return nil, err
+	}
+	for _, feed := range feedList.Items {
```

Comment:
I wonder if we could set a label and add a GetWithLabel to our client so we dont have to loop over all feeds? I dont know how involved that is so if theres some reason its a pain can we get a // TODO to make this not loop over all feeds?

Reply:
My understanding is that this client will use a local informer cache, rather than the API server, so there should be little performance penalty for iterating over all Feeds in the namespace. 

To use a label, we would essentially create a label that is identical to the OwnerReference itself. That label would have to be kept in lock step with the OwnerReference.

My preference is to use the approach where less can go wrong, as the performance hit should be very minor and only affects the control plane, not the data plane.

Thoughts?

Reply:
Fair enough - good point re: cache, the only time this should be a concern then is with a new cache but you're right the benefit isn't huge.